### ISSUE TITLE
Rename namespace to repository 

### ISSUE THREAD

**Issue by marcodelapierre** [singularityhub/singularity-hpc#378]
Just capturing (read: lazily copy-pasting) our Slack chat here.

V: for the singularity versions I added the module version to the executable name, and as you pointed out i didn't to podman's. But I'm rethinking this, and I'm wondering if in fact we shouldn't include the version? E.g., would there ever be a use case of having more than  one python (or something named the same thing) defined twice? It would be a shorter/cleaner command if we remove versions, and I can add more conflicts() clauses.

V: oh one more idea for the command naming! We could have a naming scheme that defaults to `{name}-{version}` (like with spack modules) so the person deploying the modules can decide! Maybe default would just be name though.

**Comment by marcodelapierre** [comment]
Marco: I like the idea of the naming scheme! And I think you're right, I also agree with your other points, rather than fixing the version missing in podman, it's probably more user friendly to get rid of the version in singularity. So I'd suggest `{name}`  as the default prefix.

**Comment by marcodelapierre** [comment]
Also, talking of naming conventions, I was thinking whether it'd be possible to allow a flexible naming scheme for modules, too.

E.g., say we install `quay.io/biocontainers/samtools/1.9` , I'd love being able to ditch the first part of the repo and just have `samtools/1.9` . 
I see a risk in clashing names between different recipes, so I'd probably leave the current behaviour as default. 
But for clusters that know they're not going to mix things, it'd be handy to have the option.

**Comment by vsoch** [comment]
@marcodelapierre I was thinking of this too, and the reason I haven't done it yet is because we'd need another way to identify the module once it's installed. Right now I'm just parsing the directory structure as an identifier, and I figured that since LMOD has autocompletion it wasn't a big deal to autocomplete. But I did think it would be cool to be able to do:

```bash
$ shpc install quay.io/biocontainers/samtools/1.9 samtools/1.9
```
Arguably you could do any path that you want!

```bash
$ shpc install quay.io/biocontainers/samtools/1.9 bio/samtools/1.9
```

It would be up to the admin to make sure there aren't namespace clashes, and actually we could keep the modules installed and registry files separate, but I think the main issue comes with `shpc check`, which maybe isn't a useful command, but we do match modules installed to the install folder by way of the path.

Hmm - a simple (and maybe silly idea) would be to just add the container.yaml files to the installed folder, and then derive the metadata from there? The only issue there is that the shpc check command will look at the container.yaml in the module folder, and the user would have to git pull and then shpc install again to update the module and the file. But maybe they would do that anyway and it's not a terrible idea? I'll keep thinking about it. I agree the default should be matching the namespace.

**Comment by marcodelapierre** [comment]
Can think more about this...

Right now, jut noticed there's this new line in the `settings.yaml`:
```
module_name: "{{ name }}"
```
However it's not respected in the modulefile (guess you're yet to implement it right?)

**Comment by marcodelapierre** [comment]
just found out that modules written in TCL don't like something in this new schema for aliases:
```
$ module load biocontainers/samtools/v1.9-4-deb_cv1
-bash: alias: `biocontainers/samtools:v1.9-4-deb_cv1-exec': invalid alias name
-bash: alias: `biocontainers/samtools:v1.9-4-deb_cv1-inspect-deffile': invalid alias name
-bash: alias: `biocontainers/samtools:v1.9-4-deb_cv1-inspect-runscript': invalid alias name
-bash: alias: `biocontainers/samtools:v1.9-4-deb_cv1-run': invalid alias name
-bash: alias: `biocontainers/samtools:v1.9-4-deb_cv1-shell': invalid alias name

```

**Comment by vsoch** [comment]
okay will remove slashes and replace with dash! I was doing this before and it's an oversight.

**Comment by marcodelapierre** [comment]
One more thought on alias name schema, have a look at these:
```
biocontainers/samtools:v1.9-4-deb_cv1
quay.io/biocontainers/fastqc:v123
```

Even without versions, these aliases look unnecessarily long (SUFFIX here can be exec, shell and so on):
```
biocontainers-samtools-SUFFIX
quay.io-biocontainers-fastqc-SUFFIX
```

How about this schema:
```
{{ repos }} - {{ tool }} - {{ version }}
```

`repos` here has all of the leading directories of the repository but the last one (which is `tool` instead);
default would be `{{ tool }}`  ....?

**Comment by vsoch** [comment]
This is a really good idea - let's address it after podman is merged. I'm thinking that we can have a set of regular expressions used to match module names, where some match just a name and tag, up to a complete registry, repository name, and tool, eg.,

```bash
<registry>/<repository>/<tool>
quay.io    /biocontainers/samtools
```
And then all aliases can take (as they do now) a custom format string. I think we would just default to `{{ tool }}` - so agree!

**Comment by marcodelapierre** [comment]
Oh sure you're right, agree with your latest suggestion!

**Comment by vsoch** [comment]
okay this is added to the PR! I still haven't implemented installing to any namespace that would be desired, mostly because I'm not sure this is something we would want to do given that there is tab completion already.

**Comment by vsoch** [comment]
Closing issue for now - we can bring up the need to have custom named modules if it arises.

**Comment by marcodelapierre** [comment]
Works beautifully!
Just a comment: I noticed that you used {{namespace}} for {{repository}}, and my only concern is that it could be confusing in regard to `shpc namespace`...?
Please disregard this if you don't think it's can sound ambiguous.

**Comment by vsoch** [comment]
Agree, we should change it.

**Comment by vsoch** [comment]
Closed with #397
